bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -254,12 +263,14 @@ public static HSBType rgbToHsb(PercentType[] rgb) throws IllegalArgumentExceptio
         if (r.compareTo(max) == 0) {
             hue = blueAngle.subtract(greenAngle);
         } else if (g.compareTo(max) == 0) {
-            hue = BigDecimal.valueOf(120).add(redAngle).subtract(blueAngle);
+            hue = BIG_DECIMAL_120.add(redAngle).subtract(blueAngle);
         } else {
-            hue = BigDecimal.valueOf(240).add(greenAngle).subtract(redAngle);
+            hue = BIG_DECIMAL_240.add(greenAngle).subtract(redAngle);
         }
         if (hue.compareTo(BigDecimal.ZERO) < 0) {
-            hue = hue.add(BigDecimal.valueOf(360));
+            hue = hue.add(BIG_DECIMAL_360);
+        } else if (hue.compareTo(BIG_DECIMAL_360) > 0) {
```

Comment:
@J-N-K just re-reading this line, I think it should be >= rather than > to cover the exact edge case where hue is 360. Or??

Reply:
No, I think it's fine. If that is the case 360-360=0 which is the same color.